File: server/proto/usb_descriptor.go
package proto

type UsbDescriptor struct {
	VendorName   string `json:"vendorName"`
	ProductName  string `json:"productName"`
	SerialNumber string `json:"serialNumber"`
	Vid          string `json:"vid"`
	Pid          string `json:"pid"`
}

type GetUsbDescriptorRsp struct {
	Descriptor UsbDescriptor `json:"descriptor"`
}

type SetUsbDescriptorReq struct {
	VendorName   string `json:"vendorName"`
	ProductName  string `json:"productName"`
	SerialNumber string `json:"serialNumber"`
	Vid          string `json:"vid" validate:"required"`
	Pid          string `json:"pid" validate:"required"`
}

File: web/src/api/usb-descriptor.ts
import { http } from '@/lib/http.ts';

export function getUsbDescriptor() {
  return http.get('/api/hid/usb-descriptor');
}

export function setUsbDescriptor(data: {
  vendorName: string;
  productName: string;
  serialNumber: string;
  vid: string;
  pid: string;
}) {
  return http.post('/api/hid/usb-descriptor', data);
}

export function restoreUsbDefaults() {
  return http.post('/api/hid/usb-descriptor/restore');
}
